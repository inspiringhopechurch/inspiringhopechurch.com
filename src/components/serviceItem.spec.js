import React from "react";
import { faShieldCheck, faLaptopCode } from "@fortawesome/pro-solid-svg-icons";
import { shallow } from "enzyme";
import ServiceItem from "./serviceItem";

describe("ServiceItem", () => {
  const secureDesc = (
      <>
        Applications need solid infrastructure. In our increasingly cloud-centric world we have the expertise to deploy
        your site or application quickly <strong>and</strong> <em>securely</em>
      </>
    ),
    services = [
      {
        name: "Software Development",
        icon: faLaptopCode,
        size: "2x",
        descr: `Whether it's a web application built using the latest web
    technologies, a C++ application for real-time systems, or
    customizing existing applications for your specific needs; we've
    got you covered.`,
      },
      {
        name: "Fast, Secure Hosting",
        icon: faShieldCheck,
        size: "2x",
        descr: secureDesc,
      },
    ];

  it("should render without crashing", () => {
    const wrapper = shallow(
      <ServiceItem icon={services[0].icon} serviceName={services[0].name}>
        {services[0].descr}
      </ServiceItem>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should render without crashing without icons", () => {
    const wrapper = shallow(<ServiceItem serviceName={services[1].name}>{services[1].descr}</ServiceItem>);
    expect(wrapper).toMatchSnapshot();
  });
});
