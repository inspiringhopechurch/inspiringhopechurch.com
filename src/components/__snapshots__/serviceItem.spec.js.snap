// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceItem should render without crashing 1`] = `
<div
  className="column is-half-tablet is-one-third-desktop is-one-third-fullhd"
>
  <article
    className="media"
  >
    <figure
      className="media-left"
    >
      <p
        className="image is-48x48"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                640,
                512,
                Array [],
                "f5fc",
                "M255.03 261.65c6.25 6.25 16.38 6.25 22.63 0l11.31-11.31c6.25-6.25 6.25-16.38 0-22.63L253.25 192l35.71-35.72c6.25-6.25 6.25-16.38 0-22.63l-11.31-11.31c-6.25-6.25-16.38-6.25-22.63 0l-58.34 58.34c-6.25 6.25-6.25 16.38 0 22.63l58.35 58.34zm96.01-11.3l11.31 11.31c6.25 6.25 16.38 6.25 22.63 0l58.34-58.34c6.25-6.25 6.25-16.38 0-22.63l-58.34-58.34c-6.25-6.25-16.38-6.25-22.63 0l-11.31 11.31c-6.25 6.25-6.25 16.38 0 22.63L386.75 192l-35.71 35.72c-6.25 6.25-6.25 16.38 0 22.63zM624 416H381.54c-.74 19.81-14.71 32-32.74 32H288c-18.69 0-33.02-17.47-32.77-32H16c-8.8 0-16 7.2-16 16v16c0 35.2 28.8 64 64 64h512c35.2 0 64-28.8 64-64v-16c0-8.8-7.2-16-16-16zM576 48c0-26.4-21.6-48-48-48H112C85.6 0 64 21.6 64 48v336h512V48zm-64 272H128V64h384v256z",
              ],
              "iconName": "laptop-code",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="2x"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </p>
    </figure>
    <div
      className="media-content"
    >
      <h2
        className="title is-size-3 is-uppercase"
      >
        <span
          className="has-text-weight-light"
        >
          Our
        </span>
         
        <span
          className="decorated"
        >
          Software Development
        </span>
      </h2>
      <p
        className="content is-medium"
      >
        Whether it's a web application built using the latest web
    technologies, a C++ application for real-time systems, or
    customizing existing applications for your specific needs; we've
    got you covered.
      </p>
    </div>
  </article>
</div>
`;

exports[`ServiceItem should render without crashing without icons 1`] = `
<div
  className="column is-half-tablet is-one-third-desktop is-one-third-fullhd"
>
  <article
    className="media"
  >
    <div
      className="media-content"
    >
      <h2
        className="title is-size-3 is-uppercase"
      >
        <span
          className="has-text-weight-light"
        >
          Our
        </span>
         
        <span
          className="decorated"
        >
          Fast, Secure Hosting
        </span>
      </h2>
      <p
        className="content is-medium"
      >
        Applications need solid infrastructure. In our increasingly cloud-centric world we have the expertise to deploy your site or application quickly 
        <strong>
          and
        </strong>
         
        <em>
          securely
        </em>
      </p>
    </div>
  </article>
</div>
`;
